src/ATen/native/xpu/sycl/ReflectionPadKernels.cpp

```diff
@@ -53,6 +78,118 @@ inline std::pair<int64_t, int64_t> get_index_mapping2d(
+          get_index_mapping1d(input_w_, output_w_, output_x, pad_l_, item);
+      output_data_[index_pair.second] = input_data_[index_pair.first];
+    }
+  }
+  ReflectionPad1dKernelFunctor(
+      scalar_t* input_data,
+      scalar_t* output_data,
+      int64_t input_w,
+      int64_t pad_l,
+      int64_t output_w)
+      : input_data_(input_data),
+        output_data_(output_data),
+        input_w_(input_w),
+        pad_l_(pad_l),
+        output_w_(output_w) {}
+
+ private:
+  scalar_t* input_data_;
+  scalar_t* output_data_;
+  int64_t input_w_;
+  int64_t pad_l_;
+  int64_t output_w_;
+};
+
+template <typename scalar_t>
+void reflection_pad1d_template(
+    scalar_t* input,
+    scalar_t* output,
+    int64_t input_w,
+    int64_t pad_l,
+    int64_t pad_r,
+    int64_t nbatch,
+    int64_t nplane,
+    int64_t output_w) {
+  auto queue = getCurrentSYCLQueue();
+  int64_t work_group_size = syclMaxWorkItemsPerEU();
+  int64_t work_group_num = at::ceil_div(output_w, work_group_size);
+
+  ReflectionPad1dKernelFunctor<scalar_t> kfn(
+      input, output, input_w, pad_l, output_w);
+  sycl_kernel_submit(
+      sycl::range<3>(work_group_size * work_group_num, nplane, nbatch),
+      sycl::range<3>(work_group_size, 1, 1),
+      queue,
+      kfn);
+}
+
+template <typename scalar_t>
+struct ReflectionPad1dBackwardKernelFunctor {
+  void operator()(sycl::nd_item<3> item) const {
+    auto output_x = item.get_global_id(0);
+
+    if (output_x < output_w_) {
+      // grad input index and grad output index mapping
+      auto index_pair =
+          get_index_mapping1d(input_w_, output_w_, output_x, pad_l_, item);
+      atomicAdd(
+          (sycl_global_ptr<scalar_t>)&grad_input_data_[index_pair.first],
+          grad_output_data_[index_pair.second]);
+    }
+  }
+  ReflectionPad1dBackwardKernelFunctor(
+      scalar_t* grad_input_data,
+      scalar_t* grad_output_data,
+      int64_t input_w,
+      int64_t pad_l,
+      int64_t output_w)
+      : grad_input_data_(grad_input_data),
+        grad_output_data_(grad_output_data),
+        input_w_(input_w),
+        pad_l_(pad_l),
+        output_w_(output_w) {}
+
+ private:
+  scalar_t* grad_input_data_;
+  scalar_t* grad_output_data_;
+  int64_t input_w_;
+  int64_t pad_l_;
+  int64_t output_w_;
+};
+
+template <typename scalar_t>
+void reflection_pad1d_backward_template(
+    scalar_t* grad_input,
+    scalar_t* grad_output,
+    int64_t input_w,
+    int64_t pad_l,
+    int64_t pad_r,
+    int64_t nbatch,
+    int64_t nplane,
+    int64_t output_w) {
+  auto queue = getCurrentSYCLQueue();
+  int64_t work_group_size = syclMaxWorkItemsPerEU();
+  int64_t work_group_num = at::ceil_div(output_w, work_group_size);
+
+  ReflectionPad1dBackwardKernelFunctor<scalar_t> kfn(
+      grad_input, grad_output, input_w, pad_l, output_w);
+  sycl_kernel_submit(
+      sycl::range<3>(work_group_size * work_group_num, nplane, nbatch),
+      sycl::range<3>(work_group_size, 1, 1),
```

Comment:
Please use (1, 1, work_group), though there is probably no performance issue, when highest dimension of local range is 1, in another word, it will be still coalescing for elements access on items with `get_global_id(lowest_dim)`.

Reply:
Don't forget change the kernel codes. `get_global_id(highest_dim)`